api/tasks.go

```diff
@@ -188,6 +191,127 @@ func (api *API) GetTaskHandler(w http.ResponseWriter, req *http.Request) {
 	}
 }
 
+// PutNumOfDocsHandler returns a function that updates the number_of_documents in an existing Task resource, which is associated with a specific Job.
+func (api *API) PutNumOfDocsHandler(w http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	vars := mux.Vars(req)
+	jobID := vars["job_id"]
+	taskName := vars["task_name"]
+	docCount := vars["count"]
+	logData := log.Data{"job_id": jobID, "task_name": taskName, "count": docCount}
+
+	// validate no of documents
```

Comment:
Not sure this will do much though as ```IsPositiveInteger(value int)``` is kind of same as the current ```numOfDocs < 0```, which can be tested as well. I have included unit tests for this. Please let me know.

Reply:
It just reduces the number of lines seen in handler - all the highlighted code could be put into a function and then you wouldn't need to handle error and setting the response twice - See https://github.com/ONSdigital/dp-search-reindex-api/pull/123/files#diff-3ad9ba0d601a2f9fc8d17141b39c83e0d5810f1fd653e1ab5f05ae95b0ff6008R205-R218